Upcloud: Added support for routers and gateways (#17)
elastisys/kubespray: contrib/terraform/upcloud/README.md

  * `port`: Port to load balance.
  * `target_port`: Port to the backend servers.
  * `backend_servers`: List of servers that traffic to the port should be forwarded to.

> * `connect_router`: If a router should be connected to the private network or not

Reviewer: ```suggestion
* `router_enable`: If a router should be enabled and connected to the private network or not
```

> * `gateways`: Gateways that should be connected to the router, requires connect_router is set to true

Reviewer: ```suggestion
* `gateways`: Gateways that should be connected to the router, requires router_enable is set to true
```

  * `features`: List of features for the gateway
  * `plan`: Plan to use for the gateway
  * `connections`: The connections and tunnel to create for the gateway
    * `type`: What type of connection
    * `local_routes`: Map of local routes for the connection
      * `type`: Type of route
      * `static_network`: Destination prefix of the route; needs to be a valid IPv4 prefix
    * `remote_routes`: Map of local routes for the connection
      * `type`: Type of route
      * `static_network`: Destination prefix of the route; needs to be a valid IPv4 prefix
    * `remote_address`: The remote address for the tunnel

> * `gateway_vpn_psks`: Separate variable for providing psks for connection tunnels. Environmen variabel can be exported in the following format `export TF_VAR_gateway_vpn_psks='{"${gateway-name}-${connecton-name}-tunnel":{psk:"..."}}'`

Reviewer 2: ```suggestion
* `gateway_vpn_psks`: Separate variable for providing psks for connection tunnels. Environment variable can be exported in the following format `export TF_VAR_gateway_vpn_psks='{"${gateway-name}-${connecton-name}-tunnel":{psk:"..."}}'`
```

> * `static_routes`: Static routes to apply to the router, requires `connect_router` is set to true

Reviewer: ```suggestion
* `static_routes`: Static routes to apply to the router, requires `router_enable` is set to true
```

> * `network_peerings`: Other UpCloud private networks to peer with, requires `connect_router` is set to true

Reviewer: ```suggestion
* `network_peerings`: Other UpCloud private networks to peer with, requires `router_enable` is set to true
```
